### ISSUE TITLE
pls help

### ISSUE THREAD

**Issue by dorinaaaaalin** [npm/npm#17088]
#### I'm opening this issue because:

  - [ ] npm is doing something I don't understand.

#### What's going wrong?

C:\Sites\mainserver\site>npm install module
gogo-wechat@1.0.0 C:\Sites\mainserver\site
`-- module@1.2.5
  +-- concat-stream@1.5.1
  | `-- readable-stream@2.0.6
  |   `-- string_decoder@0.10.31
  +-- lodash.template@4.2.4
  | +-- lodash._reinterpolate@3.0.0
  | +-- lodash.assigninwith@4.2.0
  | +-- lodash.keys@4.2.0
  | +-- lodash.rest@4.0.5
  | +-- lodash.templatesettings@4.1.0
  | `-- lodash.tostring@4.1.4
  +-- map-stream@0.0.6
  +-- tildify@1.2.0
  +-- vinyl-fs@2.4.3
  | +-- duplexify@3.5.0
  | | +-- end-of-stream@1.0.0
  | | | `-- once@1.3.3
  | | `-- stream-shift@1.0.0
  | +-- glob-stream@5.3.5
  | | +-- glob@5.0.15
  | | +-- glob-parent@3.1.0
  | | | `-- path-dirname@1.0.2
  | | +-- ordered-read-streams@0.3.0
  | | +-- through2@0.6.5
  | | | `-- readable-stream@1.0.34
  | | |   +-- isarray@0.0.1
  | | |   `-- string_decoder@0.10.31
  | | +-- to-absolute-glob@0.1.1
  | | | `-- extend-shallow@2.0.1
  | | `-- unique-stream@2.2.1
  | +-- gulp-sourcemaps@1.12.0
  | | +-- @gulp-sourcemaps/map-sources@1.0.0
  | | +-- acorn@4.0.13
  | | +-- css@2.2.1
  | | | +-- source-map@0.1.43
  | | | +-- source-map-resolve@0.3.1
  | | | | +-- atob@1.1.3
  | | | | +-- resolve-url@0.2.1
  | | | | `-- source-map-url@0.3.0
  | | | `-- urix@0.1.0
  | | +-- debug-fabulous@0.0.4
  | | | +-- lazy-debug-legacy@0.0.1
  | | | `-- object-assign@4.1.0
  | | +-- detect-newline@2.1.0
  | | `-- strip-bom@2.0.0
  | +-- is-valid-glob@0.3.0
  | +-- lazystream@1.0.0
  | +-- lodash.isequal@4.5.0
  | +-- merge-stream@1.0.1
  | +-- strip-bom@2.0.0
  | +-- strip-bom-stream@1.0.0
  | | +-- first-chunk-stream@1.0.0
  | | `-- strip-bom@2.0.0
  | +-- through2@2.0.3
  | +-- through2-filter@2.0.0
  | +-- vali-date@1.0.0
  | `-- vinyl@1.2.0
  |   +-- clone-stats@0.0.1
  |   `-- replace-ext@0.0.1
  `-- yargs@4.6.0
    +-- camelcase@2.1.1
    +-- cliui@3.2.0
    +-- lodash.assign@4.2.0
    +-- pkg-conf@1.1.3
    | +-- load-json-file@1.1.0
    | | `-- strip-bom@2.0.0
    | `-- symbol@0.2.3
    +-- read-pkg-up@1.0.1
    | `-- read-pkg@1.1.0
    |   +-- load-json-file@1.1.0
    |   | `-- strip-bom@2.0.0
    |   `-- path-type@1.1.0
    +-- window-size@0.2.0
    `-- yargs-parser@2.4.1
      `-- camelcase@3.0.0

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules\chokidar\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

C:\Sites\mainserver\site>npm install --no-optional

C:\Sites\mainserver\site>npm start

> gogo-wechat@1.0.0 start C:\Sites\mainserver\site
> node webpack-dev-server --progress --inline -hot

fs.js:641
  return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
                 ^

Error: ENOENT: no such file or directory, open 'C:\Sites\mainserver\site\schema\plugin.json'
    at Error (native)
    at Object.fs.openSync (fs.js:641:18)
    at Object.fs.readFileSync (fs.js:509:33)
    at validateOptions (C:\Sites\mainserver\site\node_modules\schema-utils\dist\validateOptions.js:35:27)
    at new ExtractTextPlugin (C:\Sites\mainserver\site\node_modules\extract-text-webpack-plugin\index.js:127:3)
    at Object.<anonymous> (C:\Sites\mainserver\site\webpack.config.js:10:21)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\Sites\mainserver\site\webpack-dev-server.js:8:23)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:390:7)
    at startup (bootstrap_node.js:150:9)

npm ERR! Windows_NT 10.0.15063
npm ERR! argv "C:\\node-v6.10.3-win-x64\\node.exe" "C:\\node-v6.10.3-win-x64\\node_modules\\npm\\bin\\npm-cli.js" "start"
npm ERR! node v6.10.3
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! gogo-wechat@1.0.0 start: `node webpack-dev-server --progress --inline -hot`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the gogo-wechat@1.0.0 start script 'node webpack-dev-server --progress --inline -hot'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the gogo-wechat package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node webpack-dev-server --progress --inline -hot
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs gogo-wechat
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls gogo-wechat
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Sites\mainserver\site\npm-debug.log

My package.json script

 "scripts": {
    "lint": "eslint src",
    "start": "node webpack-dev-server --progress --inline -hot",
    "build": "webpack",
    "build-prod": "cross-env NODE_ENV=production webpack -p --config ./webpack.production.config.js",
    "watch:js": "watch 'npm run build' src/",
    "build:watch": "parallelshell 'npm run watch:js'",
    "preview-dist": "http-server -a localhost -p 8000 -c-1"
  }

### supporting information:

 - `npm -v` prints: 3.10.10
 - `node -v` prints: v6.10.3
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Windows
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I access the npm registry via a VPN

**Comment by sateffen** [comment]
The "optional-dependency" is nothing to worry about, it's a macOS specific one, that gets replaced by pure js code, if not available.

The main problem, as stated in your errorlog is, that the file *C:\Sites\mainserver\site\schema\plugin.json* does not exist:

> Error: ENOENT: no such file or directory, open 'C:\Sites\mainserver\site\schema\plugin.json'

**Comment by dorinaaaaalin** [comment]
@sateffen I do not have a schema folder in site. Why is it giving me this error?

**Comment by sateffen** [comment]
That is basically the problem. Your software construct expects the file *C:\Sites\mainserver\site\schema\plugin.json* to exist. If it doesn't, you get an error. If the *schema* folder does not exist, it's the first point to evaluate, why.

It looks like your webpack build setup wants to have the file, cause it's required somewhere in your sourcecode.

**Comment by dorinaaaaalin** [comment]
@sateffen I searched schema in my site folder. The only place it comes up is in the yarn.lock file.

json-schema@0.2.3:
  version "0.2.3"
  resolved "https://registry.yarnpkg.com/json-schema/-/json-schema-0.2.3.tgz#b480c892e59a2f05954ce727bd3f2a4e882f9e13"

jsprim@^1.2.2:
  version "1.3.1"
  resolved "https://registry.yarnpkg.com/jsprim/-/jsprim-1.3.1.tgz#2a7256f70412a29ee3670aaca625994c4dcff252"
  dependencies:
    extsprintf "1.0.2"
    json-schema "0.2.3"
    verror "1.3.6"

I did the same thing with yesterday after i cloned it from a git repository. Everything worked fine. I deleted my whole folder by accident today so i did the same steps as yesterday. But this error came up. Do u think its cus of ive already did npm install and npm start cmd?
